client/van_router_create_test.go

```diff
@@ -193,10 +208,21 @@ func TestVanRouterCreateDefaults(t *testing.T) {
 		svcAccountsFound := []string{}
 
 		// Create the client
-		cli, err := newMockClient(c.namespace, "", "")
+		var cli *VanClient
+		var err error
+		if !isCluster {
+			cli, err = newMockClient(c.namespace, "", "")
+		} else {
+			cli, err = NewClient(c.namespace, "", "")
+		}
 		assert.Check(t, err, c.doc)
 
-		informers := informers.NewSharedInformerFactory(cli.KubeClient, 0)
+		if isCluster {
```

Comment:
I think you can create the namespace also in the mock and you just remove this if (the same with when you delete the namespace)